### ISSUE TITLE
namespacing, categorisation of labels?

### ISSUE THREAD

**Issue by randomm** [fatiherikli/klassify#6]
I was wondering if it was possible to easily do multiple types of categorizations (e.g. namespacing, tagging) with klassify, for it to match only from a subset of labels e.g.

    http post localhost:8888/train label="Motoring" tags="journals" text="Lorem ipsum dolor"
    http post localhost:8888/classify text="Lorem ipsum" tags="books,journals"

I'd be very tempted to use this but I need to figure out multiple categorizations from a single piece of text.

...

Perhaps this is a long winded way of saying "I'll see if I can create a patch for that" :)

**Comment by fatiherikli** [comment]
It would be really useful. I guess the training logic will not be changed. We'll just apply training process for each label.

I think we can use a list of tags instead of comma-separated tags. Klassify expecting json data on request body.

**Comment by randomm** [comment]
Yes. I'd be tempted to know from a single piece of text whether it is a journal or a book, whether it is prose or non-fiction etc etc. 

I'll see if I can get a PR together for you...

**Comment by randomm** [comment]
@fatiherikli I've done a quick and dirty backend implementation to optionally include a "context" JSON parameter for both /train and /classify calls. If the parameter is present it'll only look within that namespace. Also, when parameter is NOT present none of the namespaced data is visible to global searches.

Can you give me further info on your webpack / build setup. I'm trying to add functionality to the React UI but my webpack is failing:

`Janis-MacBook-Pro:klassify jani$ webpack
Hash: 69728e3eb83c4c6b08a9
Version: webpack 1.12.14
Time: 72ms
    Asset     Size  Chunks             Chunk Names
bundle.js  26.3 kB       0  [emitted]  main
   [0] multi main 52 bytes {0} [built] [2 errors]
    + 2 hidden modules

ERROR in multi main
Module not found: Error: Cannot resolve 'file' or 'directory' ./interface/index in /Users/jani/Sync/git/klassify
 @ multi main

ERROR in multi main
Module not found: Error: Cannot resolve module 'webpack-dev-server/client' in /Users/jani/Sync/git/klassify
 @ multi main`

If I manually alter the path to interface inside the webpack.config.js file I get further:

`ERROR in ./klassify/interface/index.js
Module parse failed: /Users/jani/Sync/git/klassify/klassify/interface/index.js Line 1: Unexpected token
You may need an appropriate loader to handle this file type.
| import React from 'react';
| import ReactDOM from 'react-dom';
| import App from './components/App';
 @ multi main`

I _think_ I've installed necessary loaders, but as you can tell I'm a noob with webpack. Nevertheless, I can create a PR for just the backend code. Would be interested in learning a bit of webpack / React also.

**Comment by fatiherikli** [comment]
I'll update the documentation for the front-end development process.

**Comment by randomm** [comment]
Great, thanks @fatiherikli !
